### ISSUE TITLE
Can not manage model relations.

### ISSUE THREAD

**Issue by ollar** [samselikoff/ember-cli-mirage#1265]
Hello friends!

I have a simple case with two models connected by one-to-many relation:

blog-post.js
```
export default DS.Model.extend({
    title: DS.attr('string'),
    body: DS.attr('string'),
    comments: DS.hasMany('comment')
});
```

comment.js
```
export default DS.Model.extend({
    author: DS.attr('string'),
    date: DS.attr('date'),
    text: DS.attr('string'),
    blogPost: DS.belongsTo('blog-post'),
});
```

Factories:
`mirage/factories/blog-post.js`
```
import { Factory, faker } from 'ember-cli-mirage';

export default Factory.extend({
    title() {
        return faker.name.title();
    },
    body() {
        return faker.lorem.text();
    },
    date() {
        return faker.date.past();
    },
});
```
`mirage/factories/comment.js`
```
import { Factory, faker, association } from 'ember-cli-mirage';

export default Factory.extend({
    author() {
        return faker.name.firstName();
    },
    date() {
        return faker.date.past();
    },
    text() {
        return faker.lorem.text();
    },
    blogPost: association(),
});
```

But when I try to generate them something is going not as I expecting:

`mirage/scenarios/default.js`

```
  let posts = server.createList('blog-post', 5);

  posts.forEach(post => {
      let comments = server.createList('comment', 2, { blogPost: post });

      post.comments = comments;
      post.save();
  })
```
Console log shows that I have 15 blog-post models and 0 comments models. Zero comments in relations also.

Here is a repo with this example https://github.com/ollar/mirage-test.

I've read docs several times, debugged magic create method, I can't sleep and eat, please help me get what am I doing wrong?

Thank you.

**Comment by geekygrappler** [comment]
https://github.com/ollar/mirage-test/blob/master/mirage/factories/comment.js#L14

I don't know what this line is, but it's wrong. Delete or comment it out and it'll work.l

**Comment by mayankbpatel** [comment]
I am experiencing this exact problem.  The repo link above is no longer active. What is on L14 that needs to be deleted?

**Comment by Mifrill** [comment]
just a soul-cry: the ember-model relations is really pain in ember-mirage :disappointed_relieved:

**Comment by samselikoff** [comment]
@Mifrill I'd love to help but probably best to open a new issue with your problem instead of posting here!

**Comment by Mifrill** [comment]
@samselikoff thanks man, I've really appreciated it, sorry about it, it's just daily weird issues, so don't worry.
But the main problem for us is:
```
"ember-cli-mirage": "0.1.14",
"ember": "2.11.3",
```
with a huge legacy code-base)

This is a good package, really, thanks for your work!

**Comment by samselikoff** [comment]
No worries, just curious to know what you're running into. I might be able to help & others might have already run into/solved what you're hitting.

What do you mean by ember-model relations?

**Comment by Mifrill** [comment]
well, thanks for this! to be specific, the last issue for example: 


the basic `belongsTo - hasMany` relation

```
author: belongsTo('user'),
```
model user:
```
...
name: attr('string'),
...
```
model post:
```
posts: hasMany('post'),
```
Then I've fetch all posts from BE:
route.js:
```
  model() {
    return this.store.findAll('post');
  },
```
now show it in hbs:
```
{{#each model as |post|}}
  post.author.name
{{/each}}
```
It's working really well in dev mode.

Now try mirage:
First of all factories - we have no `association` for this version, so the factories are empties, like this:
```
import Mirage from 'ember-cli-mirage';

export default Mirage.Factory.extend({
});
```

so, the `User hasMany Posts`, relation is required to specify ids for both records in test, like a:
```
let user = server.create('user', { name: 'Alex' });
let post = server.create('post', {
 author_id: user.id,
});
  server.db.users.update(user.id, {
    post_ids: [post.id]
  });
```
so, get error:
```
Ember Data Request GET /users/2 returned a 404
Payload (application/json)
[object Object]
```
ou yeah, mirage mock request code:
```
...
this.get('/users', { coalesce: true });
...
this.get('/posts');
```

It's not working even with alias for model, or with getting posts from user by ('user.posts')

**Comment by samselikoff** [comment]
Hm, the issue is hard to read. Can you clean up the formatting? Lot going on.

**Comment by Mifrill** [comment]
@samselikoff Thanks a ton :heart: for care! I made a simplification while describing the problem here and it helped to understand what the problem was in the test. Resolved!

well, the point is here was in a few specifics in test, in details:
we have another one model - employee.
technically it's user with type: 'employee'
`app/models/employee.js`
```
export default User.extend
```
it's necessary for login_in `signIn({ user: employee, type: 'employee' });`, nevermind.

So, the `author` in test was `employee` (server.create(:post, { author_id: employee.id })

But for some reason `server.db.users` does not contain employee, but `server.db.employees` is contains him.

So, the `post` model when we try to get `author` in `hbs` fetch the user from users - that's why we've got the error from `users` endpoint.

This is solution in specific test:
```
server.get('/users/:id', {
 users: [server.db.employees.find(employee.id)]
});
```
Thanks to you! You've force me to put the description of the issue.

The correct description of the problem - is helping to determine the reasons and it's more than half of its solution.
